lib/chef/azure/service.rb

```diff
@@ -6,9 +6,48 @@ class ChefService
   include ChefAzure::Shared
   AZURE_CHEF_SERVICE_PID_FILE = "azure-chef-client.pid"
   AZURE_CHEF_CRON_NAME = 'azure_chef_extension'
+  DEFAULT_CHEF_SERVICE_INTERVAL = 30
+  CLIENT_RB_INTERVAL_ATTRIBUTE_NAME = 'interval'
+
+  def read_client_rb(client_rb)
+    File.readlines(client_rb)
+  end
+
+  def interval_exist?(client_rb_contents)
+    client_rb_contents.any? { |line| line.include?(CLIENT_RB_INTERVAL_ATTRIBUTE_NAME + ' ') }
+  end
+
+  def interval_index(client_rb_contents)
+    client_rb_contents.index { |line| line.include?(CLIENT_RB_INTERVAL_ATTRIBUTE_NAME + ' ') }
+  end
+
+  def interval_string(chef_service_interval)
+    "#{CLIENT_RB_INTERVAL_ATTRIBUTE_NAME} #{chef_service_interval}\n"
+  end
+
+  def write_client_rb(client_rb, client_rb_contents)
+    File.write(client_rb, client_rb_contents)
+  end
+
+  def add_or_update_interval_in_client_rb(client_rb, new_chef_service_interval)
+    client_rb_contents = read_client_rb(client_rb)

```

Comment:
@mwrock yeah...here the scenarios are 

1. if `interval` attribute exist in the `client.rb` file then update it with the new value.
2. if `interval` attribute does not exist in the `client.rb` file then add a new entry for it with the given value.

So I have implemented this approach instead of using `regex` as before writing the updated contents in the `client.rb` file, I have to add checks to decide if a current entry is to be updated or a new entry has to be added.